internal/server/evaluation/evaluator.go

```diff
@@ -275,6 +200,83 @@ func (e *Evaluator) Evaluate(ctx context.Context, r *flipt.EvaluationRequest) (r
 	return resp, nil
 }
 
+// doConstraintsMatch is a utility function that will return if all or any constraints have matched for a segment depending
+// on the match type.
+func doConstraintsMatch(logger *zap.Logger, evalCtx map[string]string, constraints []storage.EvaluationConstraint, segmentMatchType flipt.MatchType) (bool, error) {
```

Comment:
`Evaluator` has as `logger` is there a reason we cant make this a method on `e *Evaluator` and use its logger instead of passing one in as an arg?

Reply:
Ah yeah no reason why. I was using an interface to just expose one method off of the Evaluator everywhere, but that is not necessary to be honest. Will change this.